src/audio/base_fw.c

```diff
@@ -334,6 +334,11 @@ static int basefw_libraries_info_get(uint32_t *data_offset, char *data)
 	return 0;
 }
 
+static int basefw_modules_info_get(uint32_t *data_offset, char *data)
+{
+	return platform_basefw_modules_info_get(data_offset, data);
```

Comment:
why it is offloaded to platform function?
What is the difference between the module info and library info (`basefw_libraries_info_get()`) to demand this handling?

Reply:
I have generally separated the function platform_basefw_modules_info_get(...) instead of implementing everything in base_fw.c because I use the ipc4_modules_info structure which uses sof_man_module.
sof_man_module is defined in rimage/manifest.h

I had a problem with the tests because on non-Intel platforms there is no manifest.h file and the tests did not pass. Besides, this modular approach follows the convention that we create code in base_fw_platform.c

With regard to the differences between basefw_modules_info_get and basefw_libraries_info_get

This basefw_libraries_info_get returns the currently loaded library and other structure. 
 basefw_modules_info_get returns which modules are loaded and which are not, along with information about them. 

In short, both commands return a different data type, different structure. In preparing this PR I looked at previous code and tests that need specific information.